core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -241,7 +241,7 @@ protected[core] object WhiskWebActionsApi extends Directives {
 
             fields.get("body") map {
                 case JsString(str) => interpretHttpResponse(code, headers, str, transid)
-                case _             => terminate(BadRequest, Messages.httpContentTypeError)(transid)
+                case js            => interpretHttpResponseAsJson(code, headers, js, transid)
```

Comment:
Those this case checks forts for js before it checks for base64 case? Making this code optimized for common case (i.e. Json string)